File: src/SmallTextButtonPartClass.rs
// Decompiled with JetBrains decompiler
// Type: WindowsApplication1.SmallTextButtonPartClass
// Assembly: WindowsApplication1, Version=1.0.8020.28903, Culture=neutral, PublicKeyToken=null
// MVID: F52869E5-0850-48AD-BBBE-68E7A4900AFE
// Assembly location: C:\Program Files (x86)\Steam\steamapps\common\Shadow Empire\ShadowEmpire.exe

// usingMicrosoft.VisualBasic;
// usingSystem;
// usingSystem.Drawing;
// usingSystem.Drawing.Drawing2D;
// usingSystem.Drawing.Imaging;
// usingSystem.Drawing.Text;

namespace WindowsApplication1
{
  pub class SmallTextButtonPartClass : SubPartClass
  {
     bool overrule;
     buttext: String;
     width: i32;
     height: i32;
     ourfont: Font;
     ourfont2: Font;
     backbitmap: Bitmap;
     bool inactive;
     bool red;
     bool tuseshadow;
     bool marcStyle;
     extraS: String;

    pub fn Click(x: i32, y: i32, let mut b: i32 = 1) -> i32
    {
      if (DrawMod.TGame.EmpireStyle)
        SoundMod.PlayAWave(DrawMod.TGame.AppPath + "sound/interface/click.wav",  DrawMod.TGame.EditObj);
      num: i32;
      return num;
    }

    pub SmallTextButtonPartClass(
      buttontext: String,
      twidth: i32,
      tDescript: String = "",
       tBackbitmap: Bitmap = null,
      let mut bbx: i32 = -1,
      let mut bby: i32 = -1,
      bool tinactive = false,
      let mut theight: i32 = 25)
      : base(twidth, theight)
    {
      self.ourfont = Font::new(DrawMod.TGame.FontCol.Families[1], 12f, FontStyle.Bold, GraphicsUnit.Pixel);
      self.overrule = false;
      self.Descript = tDescript;
      if (!Information.IsNothing( tBackbitmap))
      {
        self.backbitmap = new Bitmap(self.OwnBitmap.Width, self.OwnBitmap.Height, PixelFormat.Format32bppPArgb);
        self.backbitmap.SetResolution( DrawMod.DPIx,  DrawMod.DPIy);
        Graphics Expression = Graphics.FromImage((Image) self.backbitmap);
        Expression.CompositingMode = CompositingMode.SourceCopy;
        Expression.DrawImage((Image) tBackbitmap, Rectangle::new(0, 0, self.OwnBitmap.Width, self.OwnBitmap.Height), Rectangle::new(bbx, bby, self.OwnBitmap.Width, self.OwnBitmap.Height), GraphicsUnit.Pixel);
        Expression.CompositingMode = CompositingMode.SourceOver;
        if (!Information.IsNothing( Expression))
          Expression.Dispose();
      }
      self.buttext = buttontext;
      self.width = twidth;
      self.height = theight;
      self.inactive = tinactive;
    }

    pub fn SubDispose()
    {
      if (Information.IsNothing( self.backbitmap))
        return;
      self.backbitmap.Dispose();
      self.backbitmap = (Bitmap) null;
    }

    pub Paint: Bitmap()
    {
      SizeF sizeF1 = SizeF::new();
      Graphics Expression = Graphics.FromImage((Image) self.OwnBitmap);
      if (!Information.IsNothing( self.backbitmap))
      {
        Expression.CompositingMode = CompositingMode.SourceCopy;
        DrawMod.DrawSimple( Expression,  self.backbitmap, 0, 0);
        Expression.CompositingMode = CompositingMode.SourceOver;
      }
      Expression.SmoothingMode = SmoothingMode.AntiAlias;
      Expression.TextRenderingHint = TextRenderingHint.AntiAlias;
      Expression.TextContrast = 1;
      if (self.inactive)
      {
         let mut local1: &Graphics = &Expression;
        bitmap1: Bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
         let mut local2: &Bitmap = &bitmap1;
        Rectangle rectangle1 = Rectangle::new(7, 0, 7, 25);
        let mut srcrect1: &Rectangle = &rectangle1
        Rectangle rectangle2 = Rectangle::new(7, 0, self.width - 14, self.height);
        let mut destrect1: &Rectangle = &rectangle2
        DrawMod.DrawSimplePart2( local1,  local2, srcrect1, destrect1);
         let mut local3: &Graphics = &Expression;
        bitmap2: Bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
         let mut local4: &Bitmap = &bitmap2;
        rectangle2 = Rectangle::new(0, 0, 7, 25);
        let mut srcrect2: &Rectangle = &rectangle2
        rectangle1 = Rectangle::new(0, 0, 7, self.height);
        let mut destrect2: &Rectangle = &rectangle1
        DrawMod.DrawSimplePart2( local3,  local4, srcrect2, destrect2);
         let mut local5: &Graphics = &Expression;
        bitmap3: Bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
         let mut local6: &Bitmap = &bitmap3;
        rectangle2 = Rectangle::new(19, 0, 7, 25);
        let mut srcrect3: &Rectangle = &rectangle2
        rectangle1 = Rectangle::new(self.width - 7, 0, 7, self.height);
        let mut destrect3: &Rectangle = &rectangle1
        DrawMod.DrawSimplePart2( local5,  local6, srcrect3, destrect3);
      }
      else
      {
         let mut local7: &Graphics = &Expression;
        bitmap4: Bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALL);
         let mut local8: &Bitmap = &bitmap4;
        Rectangle rectangle3 = Rectangle::new(7, 0, 7, 25);
        let mut srcrect4: &Rectangle = &rectangle3
        Rectangle rectangle4 = Rectangle::new(7, 0, self.width - 14, self.height);
        let mut destrect4: &Rectangle = &rectangle4
        DrawMod.DrawSimplePart2( local7,  local8, srcrect4, destrect4);
         let mut local9: &Graphics = &Expression;
        bitmap5: Bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALL);
         let mut local10: &Bitmap = &bitmap5;
        rectangle3 = Rectangle::new(0, 0, 7, 25);
        let mut srcrect5: &Rectangle = &rectangle3
        rectangle4 = Rectangle::new(0, 0, 7, self.height);
        let mut destrect5: &Rectangle = &rectangle4
        DrawMod.DrawSimplePart2( local9,  local10, srcrect5, destrect5);
         let mut local11: &Graphics = &Expression;
        bitmap6: Bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALL);
         let mut local12: &Bitmap = &bitmap6;
        rectangle3 = Rectangle::new(19, 0, 7, 25);
        let mut srcrect6: &Rectangle = &rectangle3
        rectangle4 = Rectangle::new(self.width - 7, 0, 7, self.height);
        let mut destrect6: &Rectangle = &rectangle4
        DrawMod.DrawSimplePart2( local11,  local12, srcrect6, destrect6);
      }
      SizeF sizeF2 = Expression.MeasureString(self.buttext, self.ourfont);
      let mut x1: i32 =  Math.Round(2.0 + ( self.width -  sizeF2.Width) / 2.0);
      let mut y1: i32 =  Math.Round(2.0 + ( self.height -  sizeF2.Height) / 2.0);
      DrawMod.DrawTextColouredFuzzy( Expression, self.buttext, self.ourfont, x1, y1, Color.FromArgb(200, 0, 0, 0));
      SizeF sizeF3 = Expression.MeasureString(self.buttext, self.ourfont);
      let mut x2: i32 =  Math.Round(( self.width -  sizeF3.Width) / 2.0);
      let mut y2: i32 =  Math.Round(( self.height -  sizeF3.Height) / 2.0);
      DrawMod.DrawTextColouredNicely( Expression, self.buttext, self.ourfont, x2, y2, Color.White);
      if (!Information.IsNothing( Expression))
      {
        Expression.Dispose();
        Expression = (Graphics) null;
      }
      return self.OwnBitmap;
    }

    pub PaintOverlay: Bitmap()
    {
      SizeF sizeF1 = SizeF::new();
      if (self.inactive)
        return self.Paint();
      Graphics Expression = Graphics.FromImage((Image) self.OwnBitmap);
      if (!Information.IsNothing( self.backbitmap))
        DrawMod.DrawSimple( Expression,  self.backbitmap, 0, 0);
      Expression.SmoothingMode = SmoothingMode.AntiAlias;
      Expression.TextRenderingHint = TextRenderingHint.AntiAlias;
      Expression.TextContrast = 1;
       let mut local1: &Graphics = &Expression;
      bitmap: Bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
       let mut local2: &Bitmap = &bitmap;
      Rectangle rectangle1 = Rectangle::new(7, 0, 7, 25);
      let mut srcrect1: &Rectangle = &rectangle1
      Rectangle rectangle2 = Rectangle::new(7, 0, self.width - 14, self.height);
      let mut destrect1: &Rectangle = &rectangle2
      DrawMod.DrawSimplePart2( local1,  local2, srcrect1, destrect1);
       let mut local3: &Graphics = &Expression;
      bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
       let mut local4: &Bitmap = &bitmap;
      rectangle2 = Rectangle::new(0, 0, 7, 25);
      let mut srcrect2: &Rectangle = &rectangle2
      rectangle1 = Rectangle::new(0, 0, 7, self.height);
      let mut destrect2: &Rectangle = &rectangle1
      DrawMod.DrawSimplePart2( local3,  local4, srcrect2, destrect2);
       let mut local5: &Graphics = &Expression;
      bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
       let mut local6: &Bitmap = &bitmap;
      rectangle2 = Rectangle::new(19, 0, 7, 25);
      let mut srcrect3: &Rectangle = &rectangle2
      rectangle1 = Rectangle::new(self.width - 7, 0, 7, self.height);
      let mut destrect3: &Rectangle = &rectangle1
      DrawMod.DrawSimplePart2( local5,  local6, srcrect3, destrect3);
      SizeF sizeF2 = Expression.MeasureString(self.buttext, self.ourfont);
      let mut x1: i32 =  Math.Round(2.0 + ( self.width -  sizeF2.Width) / 2.0);
      let mut y1: i32 =  Math.Round(2.0 + ( self.height -  sizeF2.Height) / 2.0);
      DrawMod.DrawTextColouredFuzzy( Expression, self.buttext, self.ourfont, x1, y1, Color.FromArgb(200, 0, 0, 0));
      SizeF sizeF3 = Expression.MeasureString(self.buttext, self.ourfont);
      let mut x2: i32 =  Math.Round(( self.width -  sizeF3.Width) / 2.0);
      let mut y2: i32 =  Math.Round(( self.height -  sizeF3.Height) / 2.0);
      DrawMod.DrawTextColouredNicely( Expression, self.buttext, self.ourfont, x2, y2, Color.White);
      if (!Information.IsNothing( Expression))
        Expression.Dispose();
      return self.OwnBitmap;
    }
  }
}

File: src/SmallTextButtonPartClass.cs
// Decompiled with JetBrains decompiler
// Type: WindowsApplication1.SmallTextButtonPartClass
// Assembly: WindowsApplication1, Version=1.0.8020.28903, Culture=neutral, PublicKeyToken=null
// MVID: F52869E5-0850-48AD-BBBE-68E7A4900AFE
// Assembly location: C:\Program Files (x86)\Steam\steamapps\common\Shadow Empire\ShadowEmpire.exe

using Microsoft.VisualBasic;
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Drawing.Imaging;
using System.Drawing.Text;

namespace WindowsApplication1
{
  public class SmallTextButtonPartClass : SubPartClass
  {
    private bool overrule;
    private string buttext;
    private int width;
    private int height;
    private Font ourfont;
    private Font ourfont2;
    private Bitmap backbitmap;
    private bool inactive;
    private bool red;
    private bool tuseshadow;
    private bool marcStyle;
    private string extraS;

    public override int Click(int x, int y, int b = 1)
    {
      if (DrawMod.TGame.EmpireStyle)
        SoundMod.PlayAWave(DrawMod.TGame.AppPath + "sound/interface/click.wav", ref DrawMod.TGame.EditObj);
      int num;
      return num;
    }

    public SmallTextButtonPartClass(
      string buttontext,
      int twidth,
      string tDescript = "",
      ref Bitmap tBackbitmap = null,
      int bbx = -1,
      int bby = -1,
      bool tinactive = false,
      int theight = 25)
      : base(twidth, theight)
    {
      this.ourfont = new Font(DrawMod.TGame.FontCol.Families[1], 12f, FontStyle.Bold, GraphicsUnit.Pixel);
      this.overrule = false;
      this.Descript = tDescript;
      if (!Information.IsNothing((object) tBackbitmap))
      {
        this.backbitmap = new Bitmap(this.OwnBitmap.Width, this.OwnBitmap.Height, PixelFormat.Format32bppPArgb);
        this.backbitmap.SetResolution((float) DrawMod.DPIx, (float) DrawMod.DPIy);
        Graphics Expression = Graphics.FromImage((Image) this.backbitmap);
        Expression.CompositingMode = CompositingMode.SourceCopy;
        Expression.DrawImage((Image) tBackbitmap, new Rectangle(0, 0, this.OwnBitmap.Width, this.OwnBitmap.Height), new Rectangle(bbx, bby, this.OwnBitmap.Width, this.OwnBitmap.Height), GraphicsUnit.Pixel);
        Expression.CompositingMode = CompositingMode.SourceOver;
        if (!Information.IsNothing((object) Expression))
          Expression.Dispose();
      }
      this.buttext = buttontext;
      this.width = twidth;
      this.height = theight;
      this.inactive = tinactive;
    }

    public override void SubDispose()
    {
      if (Information.IsNothing((object) this.backbitmap))
        return;
      this.backbitmap.Dispose();
      this.backbitmap = (Bitmap) null;
    }

    public override Bitmap Paint()
    {
      SizeF sizeF1 = new SizeF();
      Graphics Expression = Graphics.FromImage((Image) this.OwnBitmap);
      if (!Information.IsNothing((object) this.backbitmap))
      {
        Expression.CompositingMode = CompositingMode.SourceCopy;
        DrawMod.DrawSimple(ref Expression, ref this.backbitmap, 0, 0);
        Expression.CompositingMode = CompositingMode.SourceOver;
      }
      Expression.SmoothingMode = SmoothingMode.AntiAlias;
      Expression.TextRenderingHint = TextRenderingHint.AntiAlias;
      Expression.TextContrast = 1;
      if (this.inactive)
      {
        ref Graphics local1 = ref Expression;
        Bitmap bitmap1 = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
        ref Bitmap local2 = ref bitmap1;
        Rectangle rectangle1 = new Rectangle(7, 0, 7, 25);
        Rectangle srcrect1 = rectangle1;
        Rectangle rectangle2 = new Rectangle(7, 0, this.width - 14, this.height);
        Rectangle destrect1 = rectangle2;
        DrawMod.DrawSimplePart2(ref local1, ref local2, srcrect1, destrect1);
        ref Graphics local3 = ref Expression;
        Bitmap bitmap2 = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
        ref Bitmap local4 = ref bitmap2;
        rectangle2 = new Rectangle(0, 0, 7, 25);
        Rectangle srcrect2 = rectangle2;
        rectangle1 = new Rectangle(0, 0, 7, this.height);
        Rectangle destrect2 = rectangle1;
        DrawMod.DrawSimplePart2(ref local3, ref local4, srcrect2, destrect2);
        ref Graphics local5 = ref Expression;
        Bitmap bitmap3 = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
        ref Bitmap local6 = ref bitmap3;
        rectangle2 = new Rectangle(19, 0, 7, 25);
        Rectangle srcrect3 = rectangle2;
        rectangle1 = new Rectangle(this.width - 7, 0, 7, this.height);
        Rectangle destrect3 = rectangle1;
        DrawMod.DrawSimplePart2(ref local5, ref local6, srcrect3, destrect3);
      }
      else
      {
        ref Graphics local7 = ref Expression;
        Bitmap bitmap4 = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALL);
        ref Bitmap local8 = ref bitmap4;
        Rectangle rectangle3 = new Rectangle(7, 0, 7, 25);
        Rectangle srcrect4 = rectangle3;
        Rectangle rectangle4 = new Rectangle(7, 0, this.width - 14, this.height);
        Rectangle destrect4 = rectangle4;
        DrawMod.DrawSimplePart2(ref local7, ref local8, srcrect4, destrect4);
        ref Graphics local9 = ref Expression;
        Bitmap bitmap5 = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALL);
        ref Bitmap local10 = ref bitmap5;
        rectangle3 = new Rectangle(0, 0, 7, 25);
        Rectangle srcrect5 = rectangle3;
        rectangle4 = new Rectangle(0, 0, 7, this.height);
        Rectangle destrect5 = rectangle4;
        DrawMod.DrawSimplePart2(ref local9, ref local10, srcrect5, destrect5);
        ref Graphics local11 = ref Expression;
        Bitmap bitmap6 = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALL);
        ref Bitmap local12 = ref bitmap6;
        rectangle3 = new Rectangle(19, 0, 7, 25);
        Rectangle srcrect6 = rectangle3;
        rectangle4 = new Rectangle(this.width - 7, 0, 7, this.height);
        Rectangle destrect6 = rectangle4;
        DrawMod.DrawSimplePart2(ref local11, ref local12, srcrect6, destrect6);
      }
      SizeF sizeF2 = Expression.MeasureString(this.buttext, this.ourfont);
      int x1 = (int) Math.Round(2.0 + ((double) this.width - (double) sizeF2.Width) / 2.0);
      int y1 = (int) Math.Round(2.0 + ((double) this.height - (double) sizeF2.Height) / 2.0);
      DrawMod.DrawTextColouredFuzzy(ref Expression, this.buttext, this.ourfont, x1, y1, Color.FromArgb(200, 0, 0, 0));
      SizeF sizeF3 = Expression.MeasureString(this.buttext, this.ourfont);
      int x2 = (int) Math.Round(((double) this.width - (double) sizeF3.Width) / 2.0);
      int y2 = (int) Math.Round(((double) this.height - (double) sizeF3.Height) / 2.0);
      DrawMod.DrawTextColouredNicely(ref Expression, this.buttext, this.ourfont, x2, y2, Color.White);
      if (!Information.IsNothing((object) Expression))
      {
        Expression.Dispose();
        Expression = (Graphics) null;
      }
      return this.OwnBitmap;
    }

    public override Bitmap PaintOverlay()
    {
      SizeF sizeF1 = new SizeF();
      if (this.inactive)
        return this.Paint();
      Graphics Expression = Graphics.FromImage((Image) this.OwnBitmap);
      if (!Information.IsNothing((object) this.backbitmap))
        DrawMod.DrawSimple(ref Expression, ref this.backbitmap, 0, 0);
      Expression.SmoothingMode = SmoothingMode.AntiAlias;
      Expression.TextRenderingHint = TextRenderingHint.AntiAlias;
      Expression.TextContrast = 1;
      ref Graphics local1 = ref Expression;
      Bitmap bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
      ref Bitmap local2 = ref bitmap;
      Rectangle rectangle1 = new Rectangle(7, 0, 7, 25);
      Rectangle srcrect1 = rectangle1;
      Rectangle rectangle2 = new Rectangle(7, 0, this.width - 14, this.height);
      Rectangle destrect1 = rectangle2;
      DrawMod.DrawSimplePart2(ref local1, ref local2, srcrect1, destrect1);
      ref Graphics local3 = ref Expression;
      bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
      ref Bitmap local4 = ref bitmap;
      rectangle2 = new Rectangle(0, 0, 7, 25);
      Rectangle srcrect2 = rectangle2;
      rectangle1 = new Rectangle(0, 0, 7, this.height);
      Rectangle destrect2 = rectangle1;
      DrawMod.DrawSimplePart2(ref local3, ref local4, srcrect2, destrect2);
      ref Graphics local5 = ref Expression;
      bitmap = BitmapStore.GetBitmap(DrawMod.TGame.BUTTONSMALLHIGH);
      ref Bitmap local6 = ref bitmap;
      rectangle2 = new Rectangle(19, 0, 7, 25);
      Rectangle srcrect3 = rectangle2;
      rectangle1 = new Rectangle(this.width - 7, 0, 7, this.height);
      Rectangle destrect3 = rectangle1;
      DrawMod.DrawSimplePart2(ref local5, ref local6, srcrect3, destrect3);
      SizeF sizeF2 = Expression.MeasureString(this.buttext, this.ourfont);
      int x1 = (int) Math.Round(2.0 + ((double) this.width - (double) sizeF2.Width) / 2.0);
      int y1 = (int) Math.Round(2.0 + ((double) this.height - (double) sizeF2.Height) / 2.0);
      DrawMod.DrawTextColouredFuzzy(ref Expression, this.buttext, this.ourfont, x1, y1, Color.FromArgb(200, 0, 0, 0));
      SizeF sizeF3 = Expression.MeasureString(this.buttext, this.ourfont);
      int x2 = (int) Math.Round(((double) this.width - (double) sizeF3.Width) / 2.0);
      int y2 = (int) Math.Round(((double) this.height - (double) sizeF3.Height) / 2.0);
      DrawMod.DrawTextColouredNicely(ref Expression, this.buttext, this.ourfont, x2, y2, Color.White);
      if (!Information.IsNothing((object) Expression))
        Expression.Dispose();
      return this.OwnBitmap;
    }
  }
}
